math.go

```diff
@@ -14,12 +17,12 @@ func dct(width int, height int, values []uint8) (result []float32) {
 				for j := 0; j < width; j++ {
 
 					sum += float64(values[i*width+j]) *
-						math.Cos(((math.Pi*float64(u))/(2*float64(height)))*(2*float64(i)+1)) *
-						math.Cos(((math.Pi*float64(v))/(2*float64(width)))*(2*float64(j)+1))
+						math.Cos(((math.Pi*float64(u))/doubleHeight)*(2*float64(i)+1)) *
+						math.Cos(((math.Pi*float64(v))/doubleWidth)*(2*float64(j)+1))
 				}
 			}
 
-			result[u*width+v] = float32(sum)
+			result[u*width+v] = int16(sum)
```

Comment:
I’m not sure how they’re representing the values—they might be using 0.0–1.0 or they might be doing some scaling (there’s lots of leeway for scaling a DCT to make it easier to handle for different purposes that doesn’t affect the result).

Here’s the greyscale values I get for the same 8x8 'A' character image they use, expressed as 8-bit signed (I think this will be nicest to work with for fingerprinting purposes):
```
  127  127  127  127  127  127  127  127
  127  127   93 -128   42  127  127  127
  127  127    8  -94  -60  127  127  127
  127  127 -111   42 -111   93  127  127
  127   42 -128 -128 -128    8  127  127
  127  -60    8  127   59 -111  127  127
   93 -128  110  127  127  -94   42  127
  127  127  127  127  127  127  127  127
```

Here’s the resulting DCT using the current function (which may benefit from some scaling; I haven’t yet checked):

```
 4439 -491 1791  215  228  395 -104   80
  318   21  459  402 -800 -447  102 -260
 1503  225 -1021 -277   80 -199  285  480
 -337  -39 -266 -292  647  357 -146  362
  396   23  125  222 -263  -75 -208 -602
   94   43 -481 -296  483  293  -29 -133
  457   55 -105  -22   -5  103 -168 -153
 -428  -63  199   65 -115 -105  192  147
```

Code used: https://gist.github.com/mandykoh/e90b72ac22c5668dcddba61d690749be